dojo/tools/npm_audit_7_plus/parser.py

```diff
@@ -0,0 +1,237 @@
+        severity = "Medium"
+    elif item_node["severity"] == "high":
+        severity = "High"
+    elif item_node["severity"] == "critical":
+        severity = "Critical"
+    else:
+        severity = "Info"
+
+    if item_node["via"] and isinstance(item_node["via"][0], str):
+        # this is a top level key (a vulnerability)
+        title = item_node["name"]
+        cwe = "CWE-1035"  # default
+        component_name = title
+        component_version = "n/a"
+
+    elif item_node["via"] and isinstance(item_node["via"][0], dict):
+        title = item_node["via"][0]["title"]
+        component_name = item_node["nodes"][0]
+        component_version = item_node["range"]
+        cwe = item_node["via"][0]["cwe"][0]
+        vuln_ids.append(cwe)
+        references.append(item_node["via"][0]["url"])
+        unique_id_from_tool = str(item_node["via"][0]["source"])
+        cvssv3 = item_node["via"][0]["cvss"]["vectorString"]
+
+    if isinstance(item_node["fixAvailable"], dict):
+        fix_name = item_node["fixAvailable"]["name"]
+        fix_version = item_node["fixAvailable"]["version"]
+        mitigation = "Update {0} to version {1}".format(fix_name, fix_version)
+    else:
+        mitigation = "No specific mitigation provided by tool."
+
+    description = get_vuln_description(item_node, tree)
+
+    if (item_node["via"] and
+        isinstance(item_node["via"][0], dict) and
+            len(item_node["via"]) > 1):
+        # we have a multiple CWE vuln which we will capture in the
+        # vulnerability_ids and references
+        for vuln in item_node["via"][1:]:  # have to decide if str or object
+            if isinstance(vuln, dict):
+                vuln_ids.append(vuln["cwe"][0])
+                references.append(vuln["url"])
+
+    if len(cwe):
+        cwe = int(cwe.split("-")[1])
+
+    dojo_finding = Finding(
+        title=title,
+        test=test,
+        severity=severity,
+        description=description,
+        cwe=cwe,
+        mitigation=mitigation,
+        references=", ".join(references),
+        component_name=component_name,
+        component_version=component_version,
+        false_p=False,
+        duplicate=False,
+        out_of_scope=False,
+        mitigated=None,
+        impact="No impact provided",
+        static_finding=static_finding,
+        dynamic_finding=False,
+        unique_id_from_tool=unique_id_from_tool,
+        vuln_id_from_tool=unique_id_from_tool
+    )
+
+    if (cvssv3 is not None) and (len(cvssv3) > 0):
+        dojo_finding.cvssv3 = cvssv3
+
+    dojo_finding.unsaved_vulnerability_ids = vuln_ids
+
+    return dojo_finding
+
+
+def get_vuln_description(item_node, tree):
+    """Make output pretty of details."""
+    effects_handled = []
+    description = ""
+
+    description = (description + item_node["name"] + " " +
+                   item_node["range"] + "\n")
+    description = description + "Severity: " + item_node["severity"] + "\n"
```

Comment:
Just a minor nit, but why not use `+=` here and in the cases below?

```suggestion
    description += item_node["name"] + " " + item_node["range"] + "\n"
    description += "Severity: " + item_node["severity"] + "\n"
```

Reply:
true, true: https://github.com/DefectDojo/django-DefectDojo/pull/9692/commits/31746fd05a45cc587dd51941ed680f5dff34a736